pkg/tools/etcd_tools.go

```diff
@@ -359,10 +409,11 @@ func (h *EtcdHelper) AtomicUpdate(key string, ptrToType runtime.Object, ignoreNo
 			return nil
 		}
 
-		_, err = h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
+		response, err := h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
```

Comment:
Blind PUT with no precondition? I don't know how important that is. Esp. since we store Spec and Status together, I don't even think we should allow that at the storage layer.